pkg/registry/clusterrolebinding/policybased/storage.go

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
I am trying to understand the escalation check logic during RoleBinding creation. If I understand it correctly, the idea here is that user cannot add themselves into a role that has more privileges that they already have. But since there is no similar check on Role update, they might as well add these extra permissions directly to one of their existing roles (if they can update roles). Is that the intended behavior?

Also, if they can't elevate their own privileges by adding a RoleBinding, but can do it for someone else, two users who can both create RoleBindings might just conspire so that one adds another to any role.

It seems that scenarios like this make the usefulness of the escalation check on RoleBinding creation somewhat limited. E.g. if someone already has enough permissions to map users to roles in a given namespace, they might as well be able to do it for themselves.